specs/consensus.md

```diff
@@ -194,6 +195,12 @@ totalCost(x, y) = x + baseCost(y) + tipCost(y)
 ```
 , where `x` above represents the amount of coins sent by the transaction authorizer and `y` above represents the a measure of the block space used by the transaction (i.e. size in bytes).
 
+Two additional helper functions are defined:
+1. `findFromQueue(power)`, which returns the address of the last validator in the [validator queue](./data_structures.md#validator) with voting power greater than or equal to `power`, or `0` if the queue is empty or no validators in the queue have at least `power` voting power.
+1. `parentFromQueue(address)`, which returns the address of the parent in the validator queue of the validator with address `address`, or `0` if `address` is not in the queue or is the head of the queue.
+
+Note that light clients cannot perform a linear search through a linked list, and are instead provided logarithmic proofs (e.g. in the case of `parentFromQueue`, a proof to the parent is provided, which should have `address` as its next validator).
+
```

Comment:
> a proof to the parent is provided, which should have `address` as its next validator

These are just Merkle proofs? I did not go through the state transition fraud proofs but a light client will need two proofs then (for the validator with `address` and a proof to the parent in the queue?

Reply:
Yes, they're just Merkle proofs, and yes, (up to) two would be needed any time a light client is executing `parentFromQueue` (e.g. in a fraud proof). Note that this isn't a different fraud proof type; it's just how the "pre-state" is given is a state transition fraud proof.